rewrite-yaml/src/main/java/org/openrewrite/yaml/MergeYaml.java

```diff
@@ -66,22 +74,98 @@ public String getDescription() {
         return "Merge a YAML snippet with an existing YAML document.";
     }
 
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getSingleSourceApplicableTest() {
+        if (StringUtils.isNullOrEmpty(filterYaml)) {
+            return null;
+        }
+
+        Yaml.Documents filter = new YamlParser().parse(filterYaml).get(0);
+        Set<YamlSearchResult> searchResults = new HashSet<>();
+
+        return new YamlVisitor<ExecutionContext>() {
+            @Override
+            public Yaml visitSequence(Yaml.Sequence sequence, ExecutionContext ctx) {
+                new ContainsTree<>(sequence, filter, key).visit(sequence, searchResults, getCursor());
+                if (doesNotMatch(searchResults)) {
+                    return sequence;
+                } else {
+                    return sequence.withMarkers(sequence.getMarkers().addIfAbsent(ContainsTree.MATCHES));
+                }
+            }
```

Comment:
Is this necessary? I haven't fully digested `ContainsTree` but I'd imagine sequences, like mappings and scalars and such, would also be covered by `visitDocuments()`. 

If this _is_ necessary, then why aren't mappings and scalars also required?

Reply:
Long story short, I think more will be as the flexibility of the filter is stretched. If you start with a mapping at the top level of the filter, then documents will work. If your first/root object of the filter is a sequence b/c the item you're referring to in the xpath is a sequence, then this is needed. For right now, I think i'll require that the xpath expression refer to a mapping and that the filterYaml always be a mapping if it exists.